pkg/codegen/dotnet/gen_program.go

```diff
@@ -233,6 +258,44 @@ func (g *generator) genTrivia(w io.Writer, token syntax.Token) {
 	}
 }
 
+func (g *generator) findFunctionSchema(function string) (*schema.Function, bool) {
+	function = LowerCamelCase(function)
+	for _, pkg := range g.program.PackageReferences() {
+		for it := pkg.Functions().Range(); it.Next(); {
+			if strings.HasSuffix(it.Token(), function) {
+				fn, err := it.Function()
+				if err != nil {
+					return nil, false
+				}
+				return fn, true
+			}
+		}
+	}
+
+	return nil, false
+}
+
+func (g *generator) isFunctionInvoke(localVariable *pcl.LocalVariable) (*schema.Function, bool) {
+	value := localVariable.Definition.Value
+	switch value.(type) {
+	case *model.FunctionCallExpression:
+		call := value.(*model.FunctionCallExpression)
+		switch call.Name {
+		case pcl.Invoke:
+			args := call.Args[0]
+			_, fullFunctionName := g.functionName(args)
+			functionNameParts := strings.Split(fullFunctionName, ".")
+			functionName := functionNameParts[len(functionNameParts)-1]
+			functionSchema, foundFunction := g.findFunctionSchema(functionName)
```

Comment:
It looks like if this check fails, we will always return `nil, false`. We can just return here. 
```suggestion
			return g.findFunctionSchema(functionName)
```

Reply:
Yeah nice catch!